code/modules/modular_computers/networking/device_types/relay.dm

```diff
@@ -21,7 +22,22 @@
 		return FALSE
 	if(!long_range && !(network_id in get_nearby_networks()))
 		return FALSE
-	return net.add_device(src)
+	. = net.add_device(src)
+	if(.)	
+		reconnect_time = null
 
+/datum/extension/network_device/broadcaster/relay/disconnect(net_down)
+	. = ..()
+	// Router went down, we should try to hook back up when it's up
+	if(net_down)
+		reconnect_time = world.time + 30 SECONDS
+		START_PROCESSING(SSprocessing, src)
+
+/datum/extension/network_device/broadcaster/relay/Process()
+	if(world.time > reconnect_time)
+		if(connect())
+			return PROCESS_KILL
```

Comment:
Does everything work fine if it somehow connected (e.g. manually) between ticks? I.e. will connect return true and do nothing if already connected?

Reply:
yeah, it checks for being connected first in that proc and returns early